[bug] Android application build failure: Failed to assemble APK

### Describe the bug

I just created a tauri project to play with it. I wanted to make an android application. I had no issues in developing the application the problem arises when it comes to building the apk. The issue here is this error arises at the end of the apk build process.  Please help

```
Execution failed for task ':app:minifyUniversalReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete, position: offset: 199, line: 7, column: 13, origin: /home/redloaded/Work/hello-tauri/src-tauri/gen/android/app/src/main/java/in/amirth24/tauriapp/generated/proguard-wry.pro

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 4s
125 actionable tasks: 7 executed, 118 up-to-date
Failed to assemble APK: command ["/home/redloaded/Work/hello-tauri/src-tauri/gen/android/gradlew", "--project-dir", "/home/redloaded/Work/hello-tauri/src-tauri/gen/android"] exited with code 1: command ["/home/redloaded/Work/hello-tauri/src-tauri/gen/android/gradlew", "--project-dir", "/home/redloaded/Work/hello-tauri/src-tauri/gen/android"] exited with code 1
    Error [tauri_cli_node] Failed to assemble APK: command ["/home/redloaded/Work/hello-tauri/src-tauri/gen/android/gradlew", "--project-dir", "/home/redloaded/Work/hello-tauri/src-tauri/gen/android"] exited with code 1: command ["/home/redloaded/Work/hello-tauri/src-tauri/gen/android/gradlew", "--project-dir", "/home/redloaded/Work/hello-tauri/src-tauri/gen/android"] exited with code 1
 ELIFECYCLE  Command failed with exit code 1.


```

With verbose flag enabled, I was able to spot this

```

Caching disabled for task ':app:minifyUniversalReleaseWithR8' because:
  Build cache is disabled
Task ':app:minifyUniversalReleaseWithR8' is not up-to-date because:
  Task has failed previously.
R8 is a new Android code shrinker. If you experience any issues, please file a bug at
https://issuetracker.google.com, using 'Shrinker (R8)' as component name.
Current version is: 8.5.27 (build 9a5bef1f7f3530a8944fd7505683ce6b7c3a7a1f from go/r8bot (luci-r8-custom-ci-archive-0-5spf)).

file or directory '/home/redloaded/Work/hello-tauri/src-tauri/gen/android/app/build/intermediates/generated_proguard_file/universalRelease/mergeUniversalReleaseGeneratedProguardFiles/proguard.txt', not found
ERROR: /home/redloaded/Work/hello-tauri/src-tauri/gen/android/app/proguard-tauri.pro:3:13: R8: Class name expected at /home/redloaded/Work/hello-tauri/src-tauri/gen/android/app/proguard-tauri.pro:3:13
-keep class `in`.amirth24.tauriapp.TauriActivity {
            ^
ERROR: /home/redloaded/Work/hello-tauri/src-tauri/gen/android/app/src/main/java/in/amirth24/tauriapp/generated/proguard-wry.pro:7:13: R8: Class name expected at /home/redloaded/Work/hello-tauri/src-tauri/gen/android/app/src/main/java/in/amirth24/tauriapp/generated/proguard-wry.pro:7:13
-keep class `in`.amirth24.tauriapp.* {
            ^
Could not execute [report metric STATISTICS_COLLECT_METRICS_OVERHEAD]
Could not execute [report metric STATISTICS_COLLECT_METRICS_OVERHEAD]

```

### Reproduction

1. Create a new tauri project with any ways I guess, I tried only with cargo and once again with pnpm.   

    - `pnpm create tauri-app`
    - `cd hello-tauri`
    - `pnpm install`

3. Initialize the android support.
    - `NDK_HOME=~/Android/Sdk/ndk/28.0.12433566 pnpm tauri android init`
    - `NDK_HOME=~/Android/Sdk/ndk/28.0.12433566 pnpm tauri android build`

### Expected behavior

_No response_

### Full `tauri info` output

```text
> hello-tauri@0.1.0 tauri /home/redloaded/Work/hello-tauri
> tauri "info"


[✔] Environment
    - OS: Pop!_OS 22.4.0 x86_64 (X64)
    ✔ webkit2gtk-4.1: 2.44.3
    ✔ rsvg2: 2.52.5
    ✔ rustc: 1.81.0 (eeb90cda1 2024-09-04)
    ✔ cargo: 1.81.0 (2dbb1af80 2024-08-20)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-unknown-linux-gnu (default)
    - node: 20.18.0
    - pnpm: 9.12.1
    - npm: 10.8.2
    - bun: 1.1.10

[-] Packages
    - tauri 🦀: 2.0.2
    - tauri-build 🦀: 2.0.1
    - wry 🦀: 0.44.1
    - tao 🦀: 0.30.3
    - @tauri-apps/api : 2.0.2
    - @tauri-apps/cli : 2.0.2

[-] Plugins
    - tauri-plugin-shell 🦀: 2.0.1
    - @tauri-apps/plugin-shell : 2.0.0

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: SolidJS
    - bundler: Vite
```


### Stack trace

_No response_

### Additional context

_No response_

--- a/.changes/android-proguard.md
+++ b/.changes/android-proguard.md
@@ -1,0 +1,7 @@
+---
+"tauri": "patch:bug"
+"tauri-cli": "patch:bug"
+"@tauri-apps/cli": "patch:bug"
+---
+
+Fix android invalid proguard file when using an `identifier` that contains a component that is a reserved kotlin keyword, like `in`, `class`, etc

--- a/crates/tauri-cli/src/mobile/android/mod.rs
+++ b/crates/tauri-cli/src/mobile/android/mod.rs
@@ -139,7 +139,7 @@ pub fn get_config(
     "WRY_ANDROID_PACKAGE",
     app.android_identifier_escape_kotlin_keyword(),
   );
-  set_var("WRY_ANDROID_PACKAGE_NO_ESCAPE", app.identifier());
+  set_var("TAURI_ANDROID_PACKAGE_UNESCAPED", app.identifier());
   set_var("WRY_ANDROID_LIBRARY", app.lib_name());
   set_var("TAURI_ANDROID_PROJECT_PATH", config.project_dir());
 

--- a/crates/tauri-cli/src/mobile/android/mod.rs
+++ b/crates/tauri-cli/src/mobile/android/mod.rs
@@ -139,6 +139,7 @@ pub fn get_config(
     "WRY_ANDROID_PACKAGE",
     app.android_identifier_escape_kotlin_keyword(),
   );
+  set_var("WRY_ANDROID_PACKAGE_NO_ESCAPE", app.identifier());
   set_var("WRY_ANDROID_LIBRARY", app.lib_name());
   set_var("TAURI_ANDROID_PROJECT_PATH", config.project_dir());
 

--- a/crates/tauri/build.rs
+++ b/crates/tauri/build.rs
@@ -267,8 +267,6 @@ fn main() {
     }
 
     if let Ok(kotlin_out_dir) = std::env::var("WRY_ANDROID_KOTLIN_FILES_OUT_DIR") {
-      let package_no_escape = env_var("WRY_ANDROID_PACKAGE_NO_ESCAPE");
-
       let package = env_var("WRY_ANDROID_PACKAGE");
       let library = env_var("WRY_ANDROID_LIBRARY");
 
@@ -298,19 +296,12 @@ fn main() {
 
         println!("cargo:rerun-if-changed={}", out_path.display());
       }
-
-      // monkey-patch for proguard-wry.pro file since wry doesn't implement
-      // logic for un/escaping the kotling keyword in identifiers
-      let wry_proguard = kotlin_out_dir.join("proguard-wry.pro");
-      let content = fs::read_to_string(&wry_proguard).expect("failed to read proguard-wry.pro");
-      fs::write(wry_proguard, content.replace(&package, &package_no_escape))
-        .expect("failed to write proguard-wry.pro");
     }
 
     if let Some(project_dir) = env::var_os("TAURI_ANDROID_PROJECT_PATH").map(PathBuf::from) {
-      let package_no_escape = env_var("WRY_ANDROID_PACKAGE_NO_ESCAPE");
+      let package_unescaped = env_var("TAURI_ANDROID_PACKAGE_UNESCAPED");
       let tauri_proguard =
-        include_str!("./mobile/proguard-tauri.pro").replace("$PACKAGE", &package_no_escape);
+        include_str!("./mobile/proguard-tauri.pro").replace("$PACKAGE", &package_unescaped);
       std::fs::write(
         project_dir.join("app").join("proguard-tauri.pro"),
         tauri_proguard,

--- a/crates/tauri/build.rs
+++ b/crates/tauri/build.rs
@@ -301,8 +301,8 @@ fn main() {
     if let Some(project_dir) = env::var_os("TAURI_ANDROID_PROJECT_PATH").map(PathBuf::from) {
       let tauri_proguard = include_str!("./mobile/proguard-tauri.pro").replace(
         "$PACKAGE",
-        &env::var("WRY_ANDROID_PACKAGE")
-          .expect("missing `WRY_ANDROID_PACKAGE` environment variable"),
+        &env::var("WRY_ANDROID_PACKAGE_NO_ESCAPE")
+          .expect("missing `WRY_ANDROID_PACKAGE_NO_ESCAPE` environment variable"),
       );
       std::fs::write(
         project_dir.join("app").join("proguard-tauri.pro"),

--- a/crates/tauri/build.rs
+++ b/crates/tauri/build.rs
@@ -260,13 +260,15 @@ fn main() {
   }
 
   if target_os == "android" {
-    if let Ok(kotlin_out_dir) = std::env::var("WRY_ANDROID_KOTLIN_FILES_OUT_DIR") {
-      fn env_var(var: &str) -> String {
-        std::env::var(var).unwrap_or_else(|_| {
-          panic!("`{var}` is not set, which is needed to generate the kotlin files for android.")
-        })
-      }
+    fn env_var(var: &str) -> String {
+      std::env::var(var).unwrap_or_else(|_| {
+        panic!("`{var}` is not set, which is needed to generate the kotlin files for android.")
+      })
+    }
 
+    let package_no_escape = env_var("WRY_ANDROID_PACKAGE_NO_ESCAPE");
+
+    if let Ok(kotlin_out_dir) = std::env::var("WRY_ANDROID_KOTLIN_FILES_OUT_DIR") {
       let package = env_var("WRY_ANDROID_PACKAGE");
       let library = env_var("WRY_ANDROID_LIBRARY");
 
@@ -296,14 +298,18 @@ fn main() {
 
         println!("cargo:rerun-if-changed={}", out_path.display());
       }
+
+      // monkey-patch for proguard-wry.pro file since wry doesn't implement
+      // logic for un/escaping the kotling keyword in identifiers
+      let wry_proguard = kotlin_out_dir.join("proguard-wry.pro");
+      let content = fs::read_to_string(&wry_proguard).expect("failed to ready proguard-wry.pro");
+      fs::write(wry_proguard, content.replace(&package, &package_no_escape))
+        .expect("failed to write proguard-wry.pro");
     }
 
     if let Some(project_dir) = env::var_os("TAURI_ANDROID_PROJECT_PATH").map(PathBuf::from) {
-      let tauri_proguard = include_str!("./mobile/proguard-tauri.pro").replace(
-        "$PACKAGE",
-        &env::var("WRY_ANDROID_PACKAGE_NO_ESCAPE")
-          .expect("missing `WRY_ANDROID_PACKAGE_NO_ESCAPE` environment variable"),
-      );
+      let tauri_proguard =
+        include_str!("./mobile/proguard-tauri.pro").replace("$PACKAGE", &package_no_escape);
       std::fs::write(
         project_dir.join("app").join("proguard-tauri.pro"),
         tauri_proguard,

--- a/crates/tauri/build.rs
+++ b/crates/tauri/build.rs
@@ -302,7 +302,7 @@ fn main() {
       // monkey-patch for proguard-wry.pro file since wry doesn't implement
       // logic for un/escaping the kotling keyword in identifiers
       let wry_proguard = kotlin_out_dir.join("proguard-wry.pro");
-      let content = fs::read_to_string(&wry_proguard).expect("failed to ready proguard-wry.pro");
+      let content = fs::read_to_string(&wry_proguard).expect("failed to read proguard-wry.pro");
       fs::write(wry_proguard, content.replace(&package, &package_no_escape))
         .expect("failed to write proguard-wry.pro");
     }

--- a/crates/tauri/build.rs
+++ b/crates/tauri/build.rs
@@ -266,9 +266,9 @@ fn main() {
       })
     }
 
-    let package_no_escape = env_var("WRY_ANDROID_PACKAGE_NO_ESCAPE");
-
     if let Ok(kotlin_out_dir) = std::env::var("WRY_ANDROID_KOTLIN_FILES_OUT_DIR") {
+      let package_no_escape = env_var("WRY_ANDROID_PACKAGE_NO_ESCAPE");
+
       let package = env_var("WRY_ANDROID_PACKAGE");
       let library = env_var("WRY_ANDROID_LIBRARY");
 
@@ -308,6 +308,7 @@ fn main() {
     }
 
     if let Some(project_dir) = env::var_os("TAURI_ANDROID_PROJECT_PATH").map(PathBuf::from) {
+      let package_no_escape = env_var("WRY_ANDROID_PACKAGE_NO_ESCAPE");
       let tauri_proguard =
         include_str!("./mobile/proguard-tauri.pro").replace("$PACKAGE", &package_no_escape);
       std::fs::write(

